Lib/sched.py

```diff
@@ -95,6 +100,17 @@ def cancel(self, event):
         with self._lock:
             self._queue.remove(event)
             heapq.heapify(self._queue)
+            if self._event:
+                self._event.set()
+
+    def cancel_all(self):
+        """Remove all events from the queue."""
+
+        with self._lock:
+            self._queue.clear()
+            heapq.heapify(self._queue)
+            if self._event:
+                self._event.set()
```

Comment:
Furthermode, this change need to be added to a `versionchanged` but for this I would wait for a core-dev if I was  you. :)